kornia/losses/total_variation.py

```diff
@@ -22,39 +27,44 @@ def total_variation(img: torch.Tensor) -> torch.Tensor:
     Reference:
         [1] https://en.wikipedia.org/wiki/Total_variation
     """
-    if not isinstance(img, torch.Tensor):
-        raise TypeError(f"Input type is not a torch.Tensor. Got {type(img)}")
-
-    if len(img.shape) < 3 or len(img.shape) > 4:
-        raise ValueError(f"Expected input tensor to be of ndim 3 or 4, but got {len(img.shape)}.")
+    KORNIA_CHECK_SHAPE(img, ["*", "H", "W"])
+    KORNIA_CHECK(reduction in ("mean", "sum"), f"Expected reduction to be one of 'mean'/'sum', but got '{reduction}'.")
 
     pixel_dif1 = img[..., 1:, :] - img[..., :-1, :]
     pixel_dif2 = img[..., :, 1:] - img[..., :, :-1]
 
-    reduce_axes = (-3, -2, -1)
-    res1 = pixel_dif1.abs().sum(dim=reduce_axes)
-    res2 = pixel_dif2.abs().sum(dim=reduce_axes)
+    res1 = pixel_dif1.abs()
+    res2 = pixel_dif2.abs()
+
+    reduce_axes = (-2, -1)
+    if reduction == "mean":
+        res1 = res1.mean(dim=reduce_axes)
+        res2 = res2.mean(dim=reduce_axes)
```

Comment:
can you extend with a small test for the different reduction cases ?

Reply:
sure